aggregator/src/test/java/org/jenkinsci/plugins/workflow/steps/TimeoutStepTest.java

```diff
@@ -27,17 +25,15 @@
     /**
      * The simplest possible timeout step ever.
      */
-    @RandomlyFails("TODO sometimes the whole build just hangs (for longer than the 10s)")
     @Test
     public void basic() throws Exception {
         WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
         p.setDefinition(new CpsFlowDefinition(
-                "node { timeout(time:5, unit:'SECONDS') { sh 'sleep 10'; echo 'NotHere' } }"));
-        WorkflowRun b = assertBuildStatus(Result.FAILURE, p.scheduleBuild2(0));
+                "node { timeout(time:5, unit:'SECONDS') { sleep 10; echo 'NotHere' } }"));
+        WorkflowRun b = assertBuildStatus(Result.ABORTED, p.scheduleBuild2(0));
```

Comment:
> why has this changed

In fact the build status was _already_ `ABORTED`, but due to a mistake in @kohsuke’s original code this was not being checked. I fixed that (later in this test), and corrected the expected status.

> as a user I would expect a timeout to be reported as failure

The `timeout` step intentionally sets the build result to `ABORTED`. Proposed changes to that behavior are off topic here.